### PR TITLE
Add Flower Datasets tests as GitHub workflow

### PR DESCRIPTION
**Pull request by adam-narozniak**
Run standard tests for Flower Dataset based on GitHub workflow.

### CODE REVIEW

File: .github/workflows/datasets.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    branches:
+      - main
+    paths:
+      - "datasets/flwr_datasets/**"
+  pull_request:
+    branches:
+      - main
+    paths:
+      - "datasets/flwr_datasets/**"
+
```
Comment (tanertopal) [inline .github/workflows/datasets.yml:14]:
```suggestion
concurrency:
  group: ${{ github.workflow }}-${{ github.ref == 'refs/heads/main' && github.run_id || github.event.pull_request.number || github.ref }}
  cancel-in-progress: true

```

File: .github/workflows/datasets.yml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+        # https://github.com/actions/runner-images/blob/main/images/linux/Ubuntu2204-Readme.md#python
+        # In case of a mismatch, the job has to download Python to install it.
+        # Note: Due to a bug in actions/setup-python we have to put 3.10 in
+        # qoutes as it will otherwise will assume 3.1
+        python: [3.8, 3.9, '3.10']
+
+    name: Python ${{ matrix.python }}
+
+    steps:
+      - uses: actions/checkout@v3
```
Comment (tanertopal) [inline .github/workflows/datasets.yml:38]:
```suggestion
      - uses: actions/checkout@v4
```

File: .github/workflows/datasets.yml [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+        # Note: Due to a bug in actions/setup-python we have to put 3.10 in
+        # qoutes as it will otherwise will assume 3.1
+        python: [3.8, 3.9, '3.10']
+
+    name: Python ${{ matrix.python }}
+
+    steps:
+      - uses: actions/checkout@v4
+      - name: Bootstrap
+        uses: ./../.github/actions/bootstrap
```
Comment (tanertopal) [inline .github/workflows/datasets.yml:40]:
```suggestion
        uses: ./.github/actions/bootstrap
```

File: .github/workflows/datasets.yml [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+  push:
+    branches:
+      - main
+    paths:
+      - "datasets/flwr_datasets/**"
+  pull_request:
+    branches:
+      - main
+    paths:
+      - "datasets/flwr_datasets/**"
```
Comment (tanertopal) [inline .github/workflows/datasets.yml:13]:
```suggestion
on:
  push:
    branches:
      - main
  pull_request:
    branches:
      - main
```

File: .github/workflows/datasets.yml [hunk 5]
```diff
@@ -0,0 +1,42 @@
+name: Flower Baselines
```
Comment (tanertopal) [inline .github/workflows/datasets.yml:1]:
```suggestion
name: Flower Datasets
```

File: .github/workflows/datasets.yml (hunk 4 above)
Comment (adam-narozniak) [reply .github/workflows/datasets.yml]: @tanertopal Why did you remove the `datasets/flwr_datasets/**`. I thought to run the tests only when there was any change in the library itself

Comment (tanertopal) [reply .github/workflows/datasets.yml]: Is their no cross depend code? Meaning if anything changes in Flower does that have any influence on datasets?

Comment (adam-narozniak) [reply .github/workflows/datasets.yml]: On the Flower's side, there'll be examples that install the Flower Datasets library (the dependency is added to the pyproject.toml). They are already implemented - it's just a modification to the existing partition code of our current examples. It can be merged when the library is released. 
On the Flower Datasets's side, I don't think we should add such tests (there will be tests that for direct ML frameworks integration).
To sum up, there should not be any case that the change of the code in one library (without a release) affects the other library.

**Review by tanertopal — APPROVED** [review]

**Review by tanertopal — APPROVED** [review]
